src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIStudyDAO.java

```diff
@@ -0,0 +1,166 @@
+        this.brAPIDAOUtil = brAPIDAOUtil;
+        this.programStudyCache = programCacheProvider.getProgramCache(this::fetchProgramStudy, BrAPIStudy.class);
+        this.brAPIEndpointProvider = brAPIEndpointProvider;
+    }
+
+    @Scheduled(initialDelay = "2s")
+    public void setup() {
+        if(!runScheduledTasks) {
+            return;
+        }
+        // Populate study cache for all programs on startup
+        log.debug("populating study cache");
+        List<Program> programs = programDAO.getActive();
+        if(programs != null) {
+            programStudyCache.populate(programs.stream().map(Program::getId).collect(Collectors.toList()));
+        }
+    }
+
+    /**
+     * Fetch the study for this program, and process it to remove storage specific values
+     * @param programId
+     * @return this program's study
+     * @throws ApiException
+     */
+    public List<BrAPIStudy> getStudies(UUID programId) throws ApiException {
+        return new ArrayList<>(programStudyCache.get(programId).values());
+    }
+
+    /**
+     * Fetch formatted study for this program
+     * @param programId
+     * @return Map<Key = string representing study UUID, value = formatted BrAPIStudy>
+     * @throws ApiException
+     */
+    private Map<String, BrAPIStudy> fetchProgramStudy(UUID programId) throws ApiException {
+        StudiesApi api = brAPIEndpointProvider.get(programDAO.getCoreClient(programId), StudiesApi.class);
+        // Get the program key
+        List<Program> programs = programDAO.get(programId);
+        if (programs.size() != 1) {
+            throw new InternalServerException("Program was not found for given key");
+        }
+        Program program = programs.get(0);
+
+        // Set query params and make call
+        BrAPIStudySearchRequest studySearch = new BrAPIStudySearchRequest();
+        studySearch.externalReferenceIDs(List.of(programId.toString()));
+        studySearch.externalReferenceSources(List.of(Utilities.generateReferenceSource(referenceSource, ExternalReferenceSource.PROGRAMS)));
+        return processStudyForDisplay(brAPIDAOUtil.search(
+                api::searchStudiesPost,
+                api::searchStudiesSearchResultsDbIdGet,
+                studySearch
+        ), program.getKey());
+    }
+
+    /**
+     * Process study into a format for display
+     * @param programStudy
+     * @return Map<Key = string representing study UUID, value = formatted BrAPIStudy>
+     * @throws ApiException
+     */
+    private Map<String,BrAPIStudy> processStudyForDisplay(List<BrAPIStudy> programStudy, String programKey) {
+        // Process the study
+        Map<String, BrAPIStudy> programStudyMap = new HashMap<>();
+        log.trace("processing germ for display: " + programStudy);
+        Map<String, BrAPIStudy> programStudyByFullName = new HashMap<>();
+        for (BrAPIStudy study: programStudy) {
+            programStudyByFullName.put(study.getStudyName(), study);
```

Comment:
`programStudyByFullName` isn't used after storing study refs and updating the studies. Can it be removed?

Reply:
Done, 47af603e.